### ISSUE TITLE
Add parsing for rollData applied to actor `movement` properties

### ISSUE THREAD

**Issue by unsoluble** [foundryvtt/dnd5e#3060]
Currently the `system.attributes.movement` object only accepts integer/null values, and can't be hit with rollData via AEs. Let's fix that. :)

**Comment by arbron** [comment]
Would probably be best to handle this alongside https://github.com/foundryvtt/dnd5e/issues/3017

The big issue is that it is difficult to predict what order these would be resolved in when one movement type is referencing another. What I've done in another system is implement a base movement value that is just a number that can then be used in formulas:

```javascript
movement: {
  base: "30",
  types: {
    walk: "@base",
    fly: "@base * 2"
  }
}
```

This would allow us to easily reference a common base speed in formulas. The base speed could also be a solution to problems like Haste which needs to double your movement, or Monk's unarmored movement which increases movement by 10.
- With haste, we have to have an active effect changing each movement type separately, which uses a lot of AE changes and becomes a problem if modules add their own movement types
- We can't implement unarmored movement because we only want it to add to movement types that don't have existing movement. With this we could add it to `movement.base` and any movement types that reference that could use it

**Comment by unsoluble** [comment]
Yeah okay, I think I see where you're going there. Not sure Haste would be as straightforward as that, as just doing `@base * 2` for it wouldn't properly double any other speed that has an addition or subtraction in its formula.

**Comment by krbz999** [comment]
Gonna assume haste is gonna be wrapped into the data prep for actors since that seems to be the trend.

Would this also solve the ability to set one movement speed to be equal to another? Such as a fathomless warlock having a swim speed equal to their walking speed?

**Comment by Fyorl** [comment]
I think we can reasonably assume that `walk` is the 'base' speed in all cases (for PCs). Sometimes the text will explicitly say 'base walking speed', but when it instead references just a character's unspecified 'speed', it always refers to the base walking speed, as far as I am aware.

**Comment by alandsidel** [comment]
If my understanding is correct, this is blocking #3505 and is itself blocked by #3017 ?  Is this planned/milestoned somewhere?
